crates/re_viewer/src/ui/view_spatial/scene/scene_part/mod.rs

```diff
@@ -51,12 +52,40 @@ pub fn instance_path_hash_for_picking<T: re_log_types::Component>(
     any_part_selected: bool,
 ) -> InstancePathHash {
     if props.interactive {
-        if entity_view.num_instances() == 1 || !any_part_selected {
-            InstancePathHash::entity_splat(ent_path)
-        } else {
-            InstancePathHash::instance(ent_path, instance_key)
-        }
+        InstancePathHash::instance(
+            ent_path,
+            instance_key_for_picking(instance_key, entity_view, any_part_selected),
+        )
     } else {
         InstancePathHash::NONE
     }
 }
+
+/// Computes the instance key that should be used for picking (in turn for selecting/hover)
+///
+/// Assumes the entity is interactive.
+///
+/// TODO(andreas): Resolve the picking upon picking instead of doing it ahead of time here. This will speed up things.
+/// Right now this is a bit hard to do since number of instances depends on the Primary. This is expected to change soon.
+pub fn instance_key_for_picking<T: re_log_types::Component>(
+    instance_key: re_log_types::component_types::InstanceKey,
+    entity_view: &re_query::EntityView<T>,
+    any_part_selected: bool,
+) -> re_log_types::component_types::InstanceKey {
+    if entity_view.num_instances() == 1 || !any_part_selected {
+        re_log_types::component_types::InstanceKey::SPLAT
```

Comment:
Some explanations would be welcome here, I have no idea what's going on?

An entity with a single instance isn't necessarily splat? But then again this logic predates this PR anyway.